internal/provider/meta.go

```diff
@@ -222,6 +232,17 @@ func NewProviderMeta(d *schema.ResourceData) (interface{}, error) {
 		}
 	}
 
+	// Set the Vault version to *ProviderMeta object
+	vaultVersionString, err := getVaultVersion(client)
```

Comment:
Would it be simpler or more consistent if `getVaultVersion`  whould return a `version.Version`?